File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort()
        start, l, ans = points[-1][0], len(points), 0
        for i in range(l-2, -1, -1):
            if points[i][1]<start:
                start = points[i][0]
                ans += 1
        return ans+1




# class Solution:
#     def findMinArrowShots(self, points: List[List[int]]) -> int:
#         points.sort()
#         st=points[-1][0]
#         ans=0
#         for i in range(len(points)-2, -1, -1):
#             if points[i][1]>=st:
#                 continue
#             else:
#                 ans+=1
#                 st=points[i][0]
#         return ans+1

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
// class Solution {
//     public int findMinArrowShots(int[][] points) {
//         Arrays.sort(points, (a,b)->a[0]-b[0]); // here a[0]-b[0] wont work because it might go outside integer limit
//         int l = points.length, start = points[l-1][0], ans = 0;
//         for(int i=l-2; i>=0; --i){
//             if(points[i][1] < start){
//                 ans++;
//                 start = points[i][0];
//             }
//         }
//         return ans+1;
//     }
// }



class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a,b) -> Integer.compare(a[0], b[0] ));
        int l = points.length, start = points[l-1][0], ans = 0;
        for(int i=l-2; i>=0; --i){
            if(points[i][1] < start){
                ans++;
                start = points[i][0];
            }
        }
        return ans+1;
    }
}